cli/program_state.rs

```diff
@@ -68,11 +72,51 @@ impl ProgramState {
     let dir = deno_dir::DenoDir::new(maybe_custom_root)?;
     let deps_cache_location = dir.root.join("deps");
     let http_cache = http_cache::HttpCache::new(&deps_cache_location);
+
+    let default_ca_stores = vec!["mozilla".to_string()];
+    let mut root_cert_store = RootCertStore::empty();
+    let ca_stores: Vec<String> = flags
+      .ca_stores
+      .clone()
+      .or_else(|| {
+        let env_ca_store = env::var("DENO_TLS_CA_STORE").ok()?;
+        Some(
+          env_ca_store
+            .split(',')
+            .map(|s| s.trim().to_string())
+            .collect(),
+        )
+      })
+      .or_else(|| Some(default_ca_stores.clone()))
+      .unwrap()
+      .iter()
+      .map(|s| s.to_string())
+      .collect();
+
+    for store in ca_stores.iter() {
+      match store.as_str() {
+        "mozilla" => {
+          root_cert_store.add_server_trust_anchors(&TLS_SERVER_ROOTS);
+        }
+        "system" => {
+          let roots = load_native_certs()
+            .expect("could not load platform certs")
+            .roots;
+          root_cert_store.roots.extend(roots);
+        }
+        _ => {
+          return Err(anyhow!("Unknown certificate store \"{}\" specified (allowed: \"system,mozilla\")", store));
+        }
+      }
+    }
+
+    // Should this do a permissions check?
     let ca_file = flags.ca_file.clone().or_else(|| env::var("DENO_CERT").ok());
-    let ca_data = match &ca_file {
-      Some(ca_file) => Some(read(ca_file).context("Failed to open ca file")?),
-      None => None,
-    };
+    if let Some(ca_file) = ca_file {
+      let certfile = File::open(&ca_file)?;
+      let mut reader = BufReader::new(certfile);
+      root_cert_store.add_pem_file(&mut reader).unwrap();
```

Comment:
I tried just changing `.unwrap()` to `?;` but for some reason it has this error:

```
error[E0277]: the trait bound `(): StdError` is not satisfied
   --> cli/program_state.rs:113:48
    |
113 |       root_cert_store.add_pem_file(&mut reader)?;
    |                                                ^ the trait `StdError` is not implemented for `()`
    |
    = note: required because of the requirements on the impl of `std::convert::From<()>` for `anyhow::Error`
    = note: required by `std::convert::From::from`
```

The docs on this function say this:

> Errors are non-specific; they may be io errors in rd and PEM format errors, but not certificate validity errors.
> This is because large collections of root certificates often include ancient or syntactically invalid certificates. CAs are competent like that.

I interpretted this to mean it basically eats the specific error and returns a default `()` which is probably useless. 

Therefore I changed it to do this:
```rs
if let Err(_err) = root_cert_store.add_pem_file(&mut reader) {
  return Err(anyhow!("Unable to add pem file to certificate store"))
}
```

@bnoordhuis If you know a more elegant one liner let me know! But I updated all of the places where this was being called like this.

Reply:
@justinmchase You could do
```rust
root_cert_store.add_pem_file(&mut reader).map_err(|_| anyhow!("Unable to add pem file to certificate store"))?
```

Reply:
`if let Err(_err) = ...` is fine, as is `.map_err(|_| ...)`.

`if let Err(()) = ...` or `.map_err(|()| ...)`  is arguably little more obvious still, otherwise a reader of the code might wonder why we're throwing away the error.